lib/web/ui/integration.go

```diff
@@ -102,3 +103,67 @@ func MakeIntegration(ig types.Integration) Integration {
 		},
 	}
 }
+
+// AWSOIDCListDatabasesRequest is a request to ListDatabases using the AWS OIDC Integration.
+type AWSOIDCListDatabasesRequest struct {
+	// RDSType is either `instance` or `cluster`.
+	RDSType string `json:"rdsType"`
+	// Engines filters the returned Databases based on their engine.
+	// Eg, mysql, postgres, mariadb, aurora, aurora-mysql, aurora-postgresql
+	Engines []string `json:"engines"`
+	// Region is the AWS Region.
+	Region string `json:"region"`
+	// NextToken is the token to be used to fetch the next page.
+	// If empty, the first page is fetched.
+	NextToken string `json:"nextToken"`
+}
+
+// MakeAWSOIDCListDatabase creates a UI representation of a Database coming from the ListDatabases request.
+func MakeAWSOIDCListDatabasesResponse(in *awsoidc.ListDatabasesResponse) *AWSOIDCListDatabasesResponse {
+	ret := &AWSOIDCListDatabasesResponse{
+		NextToken: in.NextToken,
+	}
+
+	ret.Databases = make([]AWSDatabase, 0, len(in.Databases))
+	for _, d := range in.Databases {
+		ret.Databases = append(ret.Databases, AWSDatabase{
+			Name:       d.Name,
+			Engine:     d.Engine,
+			Status:     d.Status,
+			Endpoint:   d.Endpoint,
+			Labels:     makeLabels(d.Labels),
+			ResourceID: d.ResourceID,
+			AccountID:  d.AccountID,
+		})
+	}
+
+	return ret
+}
+
+// AWSOIDCListDatabasesResponse contains a list of databases and a next token is more pages are available.
+type AWSOIDCListDatabasesResponse struct {
+	// Databases contains the page of Databases
+	Databases []AWSDatabase `json:"databases"`
+
+	// NextToken is used for pagination.
+	// If non-empty, it can be used to request the next page.
+	NextToken string `json:"nextToken,omitempty"`
+}
+
+// AWSDatabase is a representation of an AWS RDS Database
+type AWSDatabase struct {
```

Comment:
Similar question as above about these new resources, can web UI use exiting database types we already have defined? cc @kimlisa Just trying to avoid having too many representations of the same thing.

Reply:
i think it makes sense to me to be explicit like this, a type that tells me that this isn't a teleport database resource

that said, we can re-use the existing web/ui database type (linked below) and extend it to include fields unique to the aws database if we really want to (in the end it doesn't matter much as long as the front get the values we need b/c the frontend will customize it further to suite the needs of the component that needs it anyways):
https://github.com/gravitational/teleport/blob/e2df4a31ba8f810e266180f5bcba451f01bf2397/lib/web/ui/server.go#L227

no strong opinions

Reply:
I've added all the types.AWS fields and the Status as well (which comes from a label).